crates/core/src/types/transaction.rs

```diff
@@ -0,0 +1,219 @@
+use bytes::Bytes;
+use ethereum_types::{Address, H256, U256};
+use secp256k1::{ecdsa::RecoveryId, Message, SECP256K1};
+use sha3::{Digest, Keccak256};
+
+use crate::rlp::encode::RLPEncode;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Transaction {
+    LegacyTransaction(LegacyTransaction),
+    EIP1559Transaction(EIP1559Transaction),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct LegacyTransaction {
+    pub nonce: u64,
+    pub gas_price: u64,
+    pub gas: u64,
+    pub to: Address,
+    pub value: U256,
+    pub data: Bytes,
+    pub v: U256,
+    pub r: U256,
+    pub s: U256,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct EIP1559Transaction {
+    pub chain_id: u64,
+    pub signer_nonce: u64,
+    pub max_priority_fee_per_gas: u64,
+    pub max_fee_per_gas: u64,
+    pub gas_limit: u64,
+    pub destination: Address,
+    pub amount: U256,
+    pub payload: Bytes,
+    pub access_list: Vec<(Address, Vec<H256>)>,
+    pub signature_y_parity: bool,
+    pub signature_r: U256,
+    pub signature_s: U256,
+}
+
+impl RLPEncode for Transaction {
+    fn encode(&self, buf: &mut dyn bytes::BufMut) {
+        match self {
+            Transaction::LegacyTransaction(t) => t.encode(buf),
+            Transaction::EIP1559Transaction(t) => t.encode(buf),
+        };
+    }
+}
+
+impl RLPEncode for LegacyTransaction {
```

Comment:
this implementation of RLPEncode is not correct since it needs to have the length of the whole encoded tx.

For implementing this trait you can use the RLP Encoder struct. It adds the missing bits.

Reply:
This code is already on the main branch I just moved it over from the block module to the transaction module

Reply:
Ohh I see there is another PR for fixing this.